code/edgebridge/src/main/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeExtension.java

```diff
@@ -373,4 +377,65 @@ private Map<String, String> cleanContextData(Map<String, Object> eventData) {
 		}
 		return cleanedData;
 	}
+
+	/**
+	 * Adds various Analytics standard properties into the final track event payload. Creates required
+	 * parts of the hierarchy if they are not already present in the original event data. This should
+	 * be used only after first validating the event data.
+	 *
+	 * @param eventData
+	 */
+	@VisibleForTesting
+	void addAnalyticsProperties(final Map<String, Object> eventData) {
+		// Access to the `__adobe` map
+		Map<String, Object> adobeMap = DataReader.optTypedMap(
+			Object.class,
+			eventData,
+			EdgeBridgeConstants.AnalyticsKeys.ADOBE,
+			null
+		);
+		if (adobeMap == null) {
+			adobeMap = new HashMap<>();
+			eventData.put(EdgeBridgeConstants.AnalyticsKeys.ADOBE, adobeMap);
+		}
+
+		// Access to the `analytics` map
+		Map<String, Object> analyticsMap = DataReader.optTypedMap(
+			Object.class,
+			adobeMap,
+			EdgeBridgeConstants.AnalyticsKeys.ANALYTICS,
+			null
+		);
+		if (analyticsMap == null) {
+			analyticsMap = new HashMap<>();
+			adobeMap.put(EdgeBridgeConstants.AnalyticsKeys.ANALYTICS, analyticsMap);
+		}
+		// Analytics original implementation: Customer perspective defaults to foreground when unknown and is always present
+		analyticsMap.put(
+			EdgeBridgeConstants.AnalyticsKeys.CUSTOMER_PERSPECTIVE,
+			EdgeBridgeMetricsBuilder.getCustomerPerspective()
+		);
+
+		// Analytics original implementation: AppID is only populated if it passes `StringUtils.isNullOrEmpty`
+		// Note that since AppID is the only property dependent on `contextData`, it being invalid
+		// triggers an early exit here; if other metrics are added later, this early exit logic should
+		// be updated accordingly.
+		String appId = EdgeBridgeMetricsBuilder.getApplicationIdentifier();
+		if (StringUtils.isNullOrEmpty(appId)) {
+			return;
+		}
+		// Access to the `contextData` map
+		Map<String, Object> contextDataMap = DataReader.optTypedMap(
+			Object.class,
+			analyticsMap,
+			EdgeBridgeConstants.AnalyticsKeys.CONTEXT_DATA,
+			null
+		);
+		if (contextDataMap == null) {
+			contextDataMap = new HashMap<>();
+			analyticsMap.put(EdgeBridgeConstants.AnalyticsKeys.CONTEXT_DATA, contextDataMap);
+		}
```

Comment:
- the last param on the DataReader.optTypedMap is the fallback value in case the map cannot be found, you can simplify by passing new HashMap<>() instead
- same comment for the other occurrences above